File: 3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        dct = dict()
        
        i = j = r = 0
        
        while True:
            try:
                if dct.get(s[j], True):
                    dct[s[j]] = False
                    j += 1
                
                else:
                    dct[s[i]] = True
                    l = j - i
                    i += 1
                    if l > r:
                        r = l
            except:
                l = j - i
                return l if l > r else r

File: 3. Longest Substring Without Repeating Characters.cpp

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        unordered_map<char, bool> dct;
        
        int i = 0, j = 0, r = 0, l = 0 ;
        
        while (true){
            
            dct.insert({s[j], true});
            
            if ( dct[s[j]] ) {
                dct[s[j]] = false;
                if (j == s.size()) {
                    l = j - i;
                    return max(l, r);
                }
                j++;
            } else {
                dct[s[i]] = true;
                l = j - i;
                i++;
                if (l > r) {r = l;}
            }
        }
    }
};
